### PR TITLE
Add to_phased_xz_gate function for SingleQubitCliffordGate

### PR DESCRIPTION
**Pull request by ybc1991**
Allow transferring SingleQubitCliffordGate to Phased X and Z Gates.

### CODE REVIEW

**Comment by ybc1991** [comment]
In case, it is not intuitive how clifford gates can be transferred to phased xz gates, take a look at following gif as example:
Example output animation

**Review by balopat — APPROVED** [review]
LGTM with typo nits

File: cirq-core/cirq/ops/clifford_gate.py [hunk 1]
```diff
@@ -242,6 +242,61 @@ def from_unitary(u: np.ndarray) -> Optional['SingleQubitCliffordGate']:
...
+            # face rotate 120 -- (312) permutation
+            a = 0.5
+            x = 0.5 if x_to.flip else -0.5
+            z = 0.5 if x_to.flip ^ z_to.flip else -0.5
+        elif (x_to.to, z_to.to) == (pauli_gates.Y, pauli_gates.X):
+            # face rotate 240 -- (231) permutation
+            a = 0.0
+            x = -0.5 if x_to.flip ^ z_to.flip else 0.5
+            z = -0.5 if x_to.flip else 0.5
+        # It is impossible that there are other cases.
```
[the remark this answers is not in the document]
Comment (ybc1991) [reply cirq-core/cirq/ops/clifford_gate.py:296]: Thanks, that looks beter.

File: cirq-core/cirq/ops/clifford_gate.py [hunk 2]
```diff
@@ -242,6 +242,61 @@ def from_unitary(u: np.ndarray) -> Optional['SingleQubitCliffordGate']:
...
+        """Rotation of all 24 Clifford gates is isomorphic to Octahedral group.
+
+        The rotation can be catogrized by {axis} * {degree}:
+            * Identy: I
+            * {x, y, z} * {90, 180, 270}  --- {X, Y, Z} + 6 Quarter turn gates
+            * {+/-xy, +/-yz, +/-zx} * 180  --- 6 Hadamard-like gates
+            * {middle point of xyz in 4 Quadrant} * {120, 240} --- rotation in face
+        note 1 + 9 + 6 + 8 = 24 in total.
+
+        To associate with Clifford Tableau, it can also be grouped by 4:
```
[the remark this answers is not in the document]
Comment (ybc1991) [reply cirq-core/cirq/ops/clifford_gate.py:255]: The `if` branch is based on the second group while the gate name is based on the first group. 
I particularly choose the second group is for preparing to switch to the Tableau-based approach.

For example, the condition `(x_to.to, z_to.to) == (pauli_gates.X, pauli_gates.Z)` can be directly convert into something like `tableau.matrix() == np.array([[1, 0], [0, 1]])`.
Each branch condition is corresponding one of the six symplectic matrices.

File: cirq-core/cirq/ops/clifford_gate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ybc1991) [reply cirq-core/cirq/ops/clifford_gate.py]: BTW, face rotation may be ambiguous at first glance, I learn it called axis swapping rotation in some literatures.

File: cirq-core/cirq/ops/clifford_gate.py [hunk 3]
```diff
@@ -242,6 +242,62 @@ def from_unitary(u: np.ndarray) -> Optional['SingleQubitCliffordGate']:
...
+        The rotation can be categorized by {axis} * {degree}:
+            * Identity: I
+            * {x, y, z} * {90, 180, 270}  --- {X, Y, Z} + 6 Quarter turn gates
+            * {+/-xy, +/-yz, +/-zx} * 180  --- 6 Hadamard-like gates
+            * {middle point of xyz in 4 Quadrant} * {120, 240} --- swapping axis
+        note 1 + 9 + 6 + 8 = 24 in total.
+
+        To associate with Clifford Tableau, it can also be grouped by 4:
+            * {I,X,Y,Z} is [[1 0], [0, 1]]
+            * {+/- X_sqrt, 2 Hadamard-like gates acting on the XZ plane} is [[1, 0], [1, 1]]
```
Comment (balopat) [inline cirq-core/cirq/ops/clifford_gate.py:257]:
```suggestion
            * {+/- X_sqrt, 2 Hadamard-like gates acting on the XY plane} is [[1, 0], [1, 1]]
```

Comment (ybc1991) [reply cirq-core/cirq/ops/clifford_gate.py]: Sorry, I found both this one and the comment in the following Z_sqrt was wrong as well. Fixed now. (Comments under the if-else branches are correct).

File: cirq-core/cirq/ops/clifford_gate.py [hunk 4]
```diff
@@ -242,6 +242,62 @@ def from_unitary(u: np.ndarray) -> Optional['SingleQubitCliffordGate']:
...
+            x = 0.5 if x_to.flip else -0.5
+            z = 0.5 if x_to.flip ^ z_to.flip else -0.5
+        else:
+            # axis swapping rotation -- (231) permutation.
+            # This should be the only cases left.
+            assert (x_to.to, z_to.to) == (pauli_gates.Y, pauli_gates.X)
+            a = 0.0
+            x = -0.5 if x_to.flip ^ z_to.flip else 0.5
+            z = -0.5 if x_to.flip else 0.5
+
```
Comment (ybc1991) [inline cirq-core/cirq/ops/clifford_gate.py:298]: Test comments resolve

Comment (balopat) [reply cirq-core/cirq/ops/clifford_gate.py:298]: test

File: cirq-core/cirq/ops/clifford_gate.py [hunk 5]
```diff
@@ -18,7 +18,7 @@
 
 from cirq import protocols, value, linalg
 from cirq._doc import document
-from cirq.ops import common_gates, gate_features, named_qubit, pauli_gates
+from cirq.ops import common_gates, gate_features, named_qubit, pauli_gates, phased_x_z_gate
```
Comment (balopat) [inline cirq-core/cirq/ops/clifford_gate.py:21]: Resolve this.

**Pull request merged**